File: api/Entities/HR/Employment.cs
using System.ComponentModel.DataAnnotations;
using DocumentFormat.OpenXml.Wordprocessing;

namespace api.Entities.HR
{
    public class Employment: BaseEntity
    {
        public int CvRefId { get; set; }
        public int SelectionDecisionId {get; set;}
        public DateTime SelectedOn { get; set; }
        public int ChargesFixed {get; set;}
        public int Charges {get; set;}
        public string SalaryCurrency { get; set; }
        public int Salary { get; set; }
        [Range(1,36)]
        public int ContractPeriodInMonths { get; set; }=24;
        public int WeeklyHours {get; set;}=48;
        public bool HousingProvidedFree { get; set; }
        public bool HousingNotProvided { get; set; }
        public int HousingAllowance { get; set; }
        public bool FoodProvidedFree { get; set; }
        public bool FoodNotProvided { get; set; }
        public int FoodAllowance { get; set; }
        public bool TransportProvidedFree { get; set; }
        public int TransportAllowance { get; set; }
        public bool TransportNotProvided { get; set; }
        public int OtherAllowance { get; set; }
        public int LeavePerYearInDays { get; set; }
        public int LeaveAirfareEntitlementAfterMonths {get; set;}
        //public string ConclusionStatus { get; set; }
        [MaxLength(50)]
        public string OfferAccepted { get; set; }
        public DateTime OfferAcceptedOn { get; set; }
        public string OfferConclusionRegisteredByUsername { get; set; }
        [MaxLength(100)]
        public string OfferAttachmentFileName { get; set; }
        [MaxLength(250)]
        public string OfferAttachmentFullPath { get; set; }
       
    }
}

File: client/src/app/_models/admin/employment.ts
export interface IEmployment
{
	id: number;
	cvRefId: number;
	selectionDecisionId: number;

	selectedOn: Date;
	charges: number;
	salaryCurrency: string;
	salary: number;
	contractPeriodInMonths: number;
	weeklyHours: number;
	housingProvidedFree: boolean;
	housingNotProvided: boolean;
	housingAllowance: number;
	foodProvidedFree: boolean;
	foodNotProvided: boolean;
	foodAllowance: number;
	transportProvidedFree: boolean;
	transportNotProvided: boolean;
	transportAllowance: number;
	otherAllowance: number;
	leavePerYearInDays: number;
	leaveAirfareEntitlementAfterMonths: number;
	offerAccepted: string;

	categoryId: number;
	categoryName: string;
	orderItemId: number;
	orderId: number;
	orderNo: number;
	customerId: number;
	customerName: string;
	candidateId: number;
	applicationNo: number;
	candidateName: string;
	agentName: string;
	approved: boolean;
	approvedByEmpId:number;
	offerAcceptedOn: Date;
	offerAttachmentFileName: string;
	offerAttachmentFullPath: string;

}

export class Employment {
	id= 0;
	cvRefId= 0;
	selectionDecisionId= 0;
	selectedOn= new Date();
	charges= 0;
	salaryCurrency= '';
	salary= 0;
	contractPeriodInMonths= 0;
	weeklyHours= 0;
	housingProvidedFree= false;
	housingNotProvided= false;
	housingAllowance= 0;
	foodProvidedFree= false;
	foodNotProvided= false;
	foodAllowance= 0;
	transportProvidedFree= false;
	transportNotProvided= false;
	transportAllowance= 0;
	otherAllowance= 0;
	leavePerYearInDays= 0;
	leaveAirfareEntitlementAfterMonths= 0;
	offerAccepted= '';
	offerAcceptedOn= new Date();
	categoryId= 0;
	categoryName= '';
	orderItemId= 0;
	orderId= 0;
	orderNo= 0;
	customerId= 0;
	customerName= '';
	candidateId= 0;
	applicationNo= 0;
	candidateName= '';
	agentName= '';
	approved= false;
	approvedByEmpId = 0;
	offerAcceptanceConcludedOn= new Date();

}